flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -315,7 +315,11 @@ private static void deleteFileOrDirectoryInternal(File file) throws IOException
 	}
 
 	private static void deleteDirectoryInternal(File directory) throws IOException {
```

Comment:
What is happening when `directory` contains a link to another directory?

- directory
  - 1
  - symlink -> /etc/

I think that the actual issue is not `deleteDirectoryInternal` (`Files.deleteIfExists` would only delete symbolic link), but rather `cleanDirectoryInternal` not handling symbolic links correctly.

Reply:
Yes it makes more sense to check if the directory is symbolic link in `cleanDirectoryInternal `. I have updated current implementation.